agenthub/monologue_agent/utils/prompts.py

```diff
@@ -72,58 +72,73 @@
 Below is the internal monologue of an automated LLM agent. Each
 thought is an item in a JSON array. The thoughts may be memories,
 actions taken by the agent, or outputs from those actions.
-Please return a new, smaller JSON array, which summarizes the
-internal monologue. You can summarize individual thoughts, and
-you can condense related thoughts together with a description
-of their content.
+
+The monologue has two parts: the default memories, which you must not change,
+they are provided to you only for context, and the recent monologue.
+
+Please return a new, much smaller JSON array that summarizes the recent monologue.
+When summarizing, you should condense the events that appear earlier
+in the recent monologue list more aggressively, while preserving more details
+for the events that appear later in the list.
+
+You can summarize individual thoughts, and you can condense related thoughts
+together with a description of their content.
 
 %(monologue)s
 
-Make the summaries as pithy and informative as possible.
+Make the summaries as pithy and informative as possible, especially for the earlier events
+in the old monologue.
+
 Be specific about what happened and what was learned. The summary
 will be used as keywords for searching for the original memory.
 Be sure to preserve any key words or important information.
 
-Your response must be in JSON format. It must be an object with the
-key `new_monologue`, which is a JSON array containing the summarized monologue.
-Each entry in the array must have an `action` key, and an `args` key.
-The action key may be `summarize`, and `args.summary` should contain the summary.
-You can also use the same action and args from the source monologue.
+Your response must be in JSON format. It must be an object with the key `new_monologue`,
+which must be a smaller JSON array containing the summarized monologue.
+Each entry in the new monologue must have an `action` key, and an `args` key.
+You can add a summarized entry with `action` set to "summarize" and a concise summary
+in `args.summary`. You can also use the source recent event if relevant, with its original `action` and `args`.
+
+Remember you must only summarize the old monologue, not the default memories.
```

Comment:
Couldn't we simply keep the `default memories` untouched and not provide them when the summarisation is preformed?

Reply:
You think that might be better? We want the LLM to figure out how to summarize best, and the context explaining what those actions are... seemed like a good idea. Maybe it's not, we know that too much information can also confuse it... and it's an autocompletion machine, it can figure out the pattern from the recent events.

We still need some do-not-touch things, but maybe the initial thoughts of monologue are not it. Lets give it some thought!

Reply:
believe we shouldn't summarise the **ACTION_PROMPT** because doing so might result in the loss of essential information, such as required parameters for the commands. However, I am not sure or might depends on the model. It seems GPT-4-turbo was omitting stuff and putting ... etc.

Reply:
Ah, sure, we shouldn't do that. We don't send the action prompt.

I think today (main) does send the `default_events` for summarization. This branch doesn't, I was trying to tell the LLM that `default_events` are only there for context, but it should summarize only `recent events`. Is it on this branch that GPT-4 did that?

Reply:
I see. No, it is not on this branch. I haven't tried the branch yet.